lib/Compiler.cpp

```diff
@@ -757,6 +757,13 @@ int ParseOptions(const int argc, const char *const argv[]) {
     return -1;
   }
 
+  if (clspv::Option::DecorateNonUniform() &&
+      clspv::Option::SpvVersion() < clspv::Option::SPIRVVersion::SPIRV_1_5) {
+    llvm::errs()
+        << "error: NonUniform decoration is missing before SPIR-V version 1.5.";
+    return -1;
+  }
```

Comment:
The decoration is available from SPV_EXT_descriptor_indexing in earlier versions.

Reply:
Does it mean that we should enable it by default always, whatever the version. Users could disable it explicitly with a flag.
So if someone were to try to run a kernel requiring this features with an old version that does not support the extension, they would "just" have to use the flag and then hope for it to work as expected?
I would be completely ok with that, just want to make sure I understand.

Reply:
Clvk would have to check to see if it's supported, but it could reach more implementations this way. The alternative if they need the decoration and it's not supported is the kernel not working. So catching it as unsupported it probably better.